### ISSUE TITLE
Another bug viewing a thread

### ISSUE THREAD

**Issue by JimSangwine** [Foundry376/Mailspring#653]
##### What operating system are you using?
Ubuntu 17.10 & i3wm

##### What version of Mailspring are you using?
1.1.2-8ae17eae

--

**Bug?**

I get the following error trying to view a thread:

```
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.extname (path.js:1452:5)
    at Object.shouldDisplayAsImage (/src/flux/models/utils.es6:170:22)
    at files.filter.f (/internal_packages/attachments/lib/message-attachments.jsx:79:52)
    at Array.filter (native)
    at MessageAttachments.render (/internal_packages/attachments/lib/message-attachments.jsx:79:33)
    at h (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:130:280)
    at beginWork (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:133:485)
    at d (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:161:93)
    at f (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:161:417)
    at g (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:162:149)
    at m (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169:173)
    at w (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:168:413)
    at z (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:167:32)
    at u (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:165:441)
    at Object.enqueueSetState (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:108:299)
    at MessageList.A.setState (/snap/mailspring/140/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.production.min.js:12:357)
    at MessageStore.MessageList._onChange (/internal_packages/message-list/lib/message-list.jsx:353:10)
    at EventEmitter.eventHandler (/src/global/mailspring-store.es6:125:23)
    at emitOne (events.js:120:20)
    at EventEmitter.emit (events.js:210:7)
    at MessageStore.trigger (/src/global/mailspring-store.es6:137:26)
    at query.then.items (/src/flux/stores/message-store.es6:293:12)
    at <anonymous>
```
I have only ever viewed or replied to this thread in Mailspring.

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

Well my account is a Gmail one but I don't know if it's related.

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Sorry - no idea how to reproduce this.

--

**Feature Request?**
##### Does this feature exist in another mail client or tool you use?

I checked the thread in Gmail - and (at least) the last couple of messages are missing!! 
I'm going to have to switch away from Mailspring until it's a little more stable - that was a really important thread and I'm concerned I might have missed a reply :(

**Comment by bengotow** [comment]
Hey! Thanks for reporting this—this is fixed in 1.1.3! If you haven't received the update yet, you can download it manually via getmailspring.com/download.  If you're using the `snap` release on Ubuntu, you might just need to `snap refresh mailspring`. Hope that helps!

**Comment by JimSangwine** [comment]
Thanks @bengotow ,

The thread does seem viewable now.
I still seem to be missing a couple of messages from it though - is this possible?
